core/src/main/scala/com/ubirch/user/core/actor/UserActorLogic.scala

```diff
@@ -0,0 +1,74 @@
+package com.ubirch.user.core.actor
+
+import com.ubirch.user.core.manager.UserManager
+import com.ubirch.user.model.db.{Activate, Deactivate, User}
+import com.ubirch.user.model.rest.ActivationUpdate
+import com.ubirch.user.model.rest.ActivationUpdate._
+import com.ubirch.util.date.DateUtil
+import com.ubirch.util.mongo.connection.MongoUtil
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+trait UserActorLogic {
+
+  protected[actor] def updateActivation(updates: ActivationUpdate)(implicit mongo: MongoUtil): Future[Either[String, String]] = {
+
+    val activationMap = updates.updates.map(u => u.externalId -> (u.activate, u.executionDate)).toMap
+
+    for {
+      users <- UserManager.findByExternalIds(activationMap.keys.toSeq)
+      result <- validate(updates, users).flatMap {
+        case Left(errorCsv) => Future.successful(Left(errorCsv))
+        case _ => updateUsers(updates, users).map(successMsg => Right(header + lineBreak + successMsg.mkString(lineBreak)))
+      }
+    } yield result
+
+  }
+
+  private[actor] def validate(update: ActivationUpdate, users: Seq[User]): Future[Either[String, Unit]] = {
+
+    val now = DateUtil.nowUTC
+
+    val errors = update.updates.map { u =>
+      users.find(_.externalId == u.externalId) match {
+        case _ if u.executionDate.isDefined && u.executionDate.get.isBefore(now) =>
```

Comment:
why don't you call here before this case-match?
calling `users.find(_.externalId == u.externalId)` looks unnecessary in this case
Here `None` and `Some(_)` with this condition can pass here.